glean-core/android/src/main/java/mozilla/telemetry/glean/private/BooleanMetricType.kt

```diff
@@ -14,16 +14,31 @@ import mozilla.telemetry.glean.rust.RustError
 import mozilla.telemetry.glean.Dispatchers
 import mozilla.telemetry.glean.rust.toBoolean
 
-class BooleanMetricType(
-    disabled: Boolean,
-    category: String,
-    lifetime: Lifetime,
-    name: String,
+/**
+ * This implements the developer facing API for recording boolean metrics.
+ *
+ * Instances of this class type are automatically generated by the parsers at build time,
+ * allowing developers to record values that were previously registered in the metrics.yaml file.
+ *
+ * The boolean API only exposes the [set] method.
+ *
+ * The internal constructor is only used by labeled metrics directly.
+ */
+class BooleanMetricType internal constructor(
+    var handle: Long,
     val sendInPings: List<String>
 ) {
-    private var handle: Long
 
-    init {
+    /**
+     * The public constructor used by automatically generated metrics.
+     */
+    constructor(
+        disabled: Boolean,
+        category: String,
+        lifetime: Lifetime,
+        name: String,
+        sendInPings: List<String>
+    ) : this(0, sendInPings) {
```

Comment:
nit: since we've been bitten by this before in glean-ac, can you use named arguments here, i.e. `this(handle = 0, sendInPings = sendInPings)`?